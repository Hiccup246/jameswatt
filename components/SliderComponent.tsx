import { ReactElement } from "react";

export default function SliderComponent({
  children,
  slideRight,
}: {
  children: ReactElement[];
  slideRight?: boolean;
}) {
  const slideAnimation = !!slideRight
    ? "animate-scroll-reverse"
    : "animate-scroll";

  return (
    <div className="container relative w-full overflow-hidden">
      <div className="relative m-auto h-fit w-full">
        <div
          className={`slide-track hover:animation-pause flex w-fit ${slideAnimation}`}
        >
          {children}
        </div>
      </div>
    </div>
  );
}
