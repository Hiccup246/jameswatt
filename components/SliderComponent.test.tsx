import SliderComponent from "./SliderComponent";
import renderer from "react-test-renderer";

describe("SliderComponent", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <SliderComponent>
          <div>Div 1</div>
          <div>Div 2</div>
        </SliderComponent>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
