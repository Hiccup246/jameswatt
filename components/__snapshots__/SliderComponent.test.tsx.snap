// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderComponent renders correctly 1`] = `
<div
  className="container relative w-full overflow-hidden"
>
  <div
    className="relative m-auto h-fit w-full"
  >
    <div
      className="slide-track hover:animation-pause flex w-fit animate-scroll-v2"
    >
      <div>
        Div 1
      </div>
      <div>
        Div 2
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the SliderComponent correctly 1`] = `
<div
  className="container relative w-full overflow-hidden"
>
  <div
    className="relative m-auto h-fit w-full"
  >
    <div
      className="slide-track hover:animation-pause flex w-fit animate-scroll-v2"
    >
      <div>
        Div 1
      </div>
      <div>
        Div 2
      </div>
    </div>
  </div>
</div>
`;
